examples/src/main/scala/org/apache/spark/examples/mllib/PowerIterationClusteringExample.scala

```diff
@@ -0,0 +1,149 @@
+      opt[Int]('k', "k")
+          .text(s"number of circles (/clusters), default: ${defaultParams.k}")
+          .action((x, c) => c.copy(k = x))
+      opt[Int]('n', "n")
+          .text(s"number of points, default: ${defaultParams.numPoints}")
+          .action((x, c) => c.copy(numPoints = x))
+      opt[Int]("numIterations")
+          .text(s"number of iterations, default: ${defaultParams.numIterations}")
+          .action((x, c) => c.copy(numIterations = x))
+    }
+
+    parser.parse(args, defaultParams).map { params =>
+      run(params)
+    }.getOrElse {
+      sys.exit(1)
+    }
+  }
+
+  def run(params: Params) {
+    val conf = new SparkConf()
+        .setMaster("local")
+        .setAppName(s"PowerIterationClustering with $params")
+    val sc = new SparkContext(conf)
+
+    Logger.getRootLogger.setLevel(Level.WARN)
+
+    val circlesRdd = generateCirclesRdd(sc, params.k, params.numPoints, params.outerRadius)
+    val model = new PowerIterationClustering()
+        .setK(params.k)
+        .setMaxIterations(params.numIterations)
+        .run(circlesRdd)
+
+    val clusters = model.assignments.collect.groupBy(_._2).mapValues(_.map(_._1))
+    val assignments = clusters.toList.sortBy { case (k, v) => v.length}
+    val assignmentsStr = assignments
+        .map { case (k, v) => s"$k -> ${v.sorted.mkString("[", ",", "]")}"}.mkString(",")
+    println(s"Cluster assignments: $assignmentsStr")
+
+    sc.stop()
+  }
+
+  def generateCirclesRdd(sc: SparkContext,
+      nCircles: Int = 3,
+      nPoints: Int = 30,
+      outerRadius: Double): RDD[(Long, Long, Double)] = {
+
+    val radii = for (cx <- 0 until nCircles) yield outerRadius / (nCircles-cx)
+    val groupSizes = for (cx <- 0 until nCircles) yield (cx + 1) * nPoints
+    var ix = 0
+    val points = for (cx <- 0 until nCircles;
+                      px <- 0 until groupSizes(cx)) yield {
+      val theta = 2.0 * math.Pi * px / groupSizes(cx)
+      val out = (ix, (radii(cx) * math.cos(theta), radii(cx) * math.sin(theta)))
+      ix += 1
+      out
+    }
+    val rdd = sc.parallelize(points)
+    val distancesRdd = rdd.cartesian(rdd).flatMap { case ((i0, (x0, y0)), (i1, (x1, y1))) =>
+      if (i0 < i1) {
+        Some((i0.toLong, i1.toLong, similarity((x0, y0), (x1, y1))))
+      } else {
+        None
+      }
+    }
+    distancesRdd
+  }
+
+  private[mllib] def similarity(p1: (Double, Double), p2: (Double, Double)) = {
+    gaussianSimilarity(p1, p2, 1.0)
+  }
+
+  /**
+   * Gaussian Similarity:  http://en.wikipedia.org/wiki/Radial_basis_function_kernel
+   */
+  def gaussianSimilarity(p1: (Double, Double), p2: (Double, Double), sigma: Double) = {
+    math.exp((p1._1 - p2._1) * (p1._1 - p2._1) + (p1._2 - p2._2) * (p1._2 - p2._2))
+  }
+
+  case class Params(
+      input: String = null,
+      k: Int = 3,
+      numPoints: Int = 5,
+      numIterations: Int = 10,
+      outerRadius: Double = 3.0
+      ) extends AbstractParams[Params]
```

Comment:
Merge this line to the one above. Maybe it is better to move `Params` to the beginning of `object PIC` because it contains the default values, which are important. No strong preference.